src/renderer/components/uielements/fees/Fees.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useRef, useMemo } from 'react'
+
+import * as RD from '@devexperts/remote-data-ts'
+import { BaseAmount, Asset } from '@thorchain/asgardex-util'
+import * as FP from 'fp-ts/function'
+import * as A from 'fp-ts/lib/Array'
+import { useIntl } from 'react-intl'
+
+import { formatFee } from './Fees.helper'
+
+type Fee = {
+  amount: BaseAmount
+  asset: Asset
+}
+
+type Props = {
+  fees: RD.RemoteData<Error, Fee[]>
+  hasCrossChainFee?: boolean
+}
+
+export const Fees: React.FC<Props> = ({ fees, hasCrossChainFee }) => {
+  const intl = useIntl()
+
+  const prevFeesRef = useRef<string>('...')
+
+  const feesFormattedValue = useMemo(
+    () =>
+      FP.pipe(
+        fees,
+        RD.map(
+          FP.flow(
+            A.map(formatFee),
+            A.reduce('', (acc, cur) => `${acc}${acc ? ' + ' : ' '}${cur}`)
+          )
+        ),
+        RD.fold(
+          () => <>{prevFeesRef.current}</>,
+          () => <>{prevFeesRef.current}</>,
+          (error) => (
+            <>
+              {intl.formatMessage({ id: 'common.error' })}: {error.message}
+            </>
+          ),
+          (fees) => {
+            prevFeesRef.current = fees
+            return <>{fees}</>
+          }
+        )
+      ),
+    [fees, intl]
+  )
+
+  return (
+    <>
+      {intl.formatMessage({ id: hasCrossChainFee ? 'common.fees' : 'common.fee' })}:{feesFormattedValue}
```

Comment:
`hasCrossChainFee` -> fees.length > 1

Reply:
not at all. fees is RD of Array not Array of RDs. also if we get folded value of inner array type there might be inconsistent in case of pending/success states

Reply:
I still think we don't need a `hasCrossChainFee` property. The component can check itself if there is one or more fees by checking the result of fees . In case of loading / error state just show `common.fee`. It will simplify things for using this component